mlflow/gateway/client.py

```diff
@@ -95,26 +96,25 @@ def get_route(self, name: str):
 
         return Route(**response)
 
-    def search_routes(self, search_filter: Optional[str] = None):
+    def search_routes(self, page_token: Optional[str] = None) -> PagedList[Route]:
         """
-        Search for routes in the Gateway. Currently, this simply returns all configured routes.
+        Search for routes in the Gateway.
 
-        .. note::
-            Search is currently not implemented. Providing a `search_filter` term will raise an
-            exception. Leave the arguments empty to get a listing of all configured routes.
-
-        :param search_filter: An optional filter to apply to the search. Currently not used.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                           a prior ``search_routes()`` call.
         :return: Returns a list of all configured and initialized `Route` data for the MLflow
             Gateway Server. The return will be a list of dictionaries that detail the name, type,
             and model details of each active route endpoint.
         """
-        if search_filter:
-            raise MlflowException.invalid_parameter_value(
-                "Search functionality is not implemented. This API only returns all configured "
-                "routes with no `search_filter` defined."
-            )
-        response = self._call_endpoint("GET", MLFLOW_GATEWAY_CRUD_ROUTE_BASE).json()["routes"]
-        return [Route(**resp) for resp in response]
+        request_parameters = None
+        if page_token is not None:
+            request_parameters = {"page_token": page_token}
+        response_json = self._call_endpoint(
+            "GET", MLFLOW_GATEWAY_CRUD_ROUTE_BASE, json_body=json.dumps(request_parameters)
+        ).json()
```

Comment:
/ai what does `json.dumps(None)` returns in Python?

Reply:
Would rather not pass the null field. Let me know if you strongly disagree

Reply:
Can we pass `None` to `json_body` if `request_parameters` is None?

Reply:
Ah, yes - fixed